lib/models/WebsiteConfiguration.ts

```diff
@@ -0,0 +1,215 @@
+export type Redirect = {
+    protocol?: string;
+    hostName?: string;
+    replaceKeyPrefixWith?: string;
+    replaceKeyWith?: string;
+    httpRedirectCode: string;
+};
+export type Condition = {
+    keyPrefixEquals?: string;
+    httpErrorCodeReturnedEquals?: string;
+};
+export type RoutingRuleParams = { redirect: Redirect; condition?: Condition };
+
+export class RoutingRule {
+    _redirect?: Redirect;
+    _condition?: Condition;
+
+    /**
+    * Represents a routing rule in a website configuration.
+    * @constructor
+    * @param params - object containing redirect and condition objects
+    * @param params.redirect - specifies how to redirect requests
+    * @param [params.redirect.protocol] - protocol to use for redirect
+    * @param [params.redirect.hostName] - hostname to use for redirect
+    * @param [params.redirect.replaceKeyPrefixWith] - string to replace
+    *   keyPrefixEquals specified in condition
+    * @param [params.redirect.replaceKeyWith] - string to replace key
+    * @param [params.redirect.httpRedirectCode] - http redirect code
+    * @param [params.condition] - specifies conditions for a redirect
+    * @param [params.condition.keyPrefixEquals] - key prefix that
+    *   triggers a redirect
+    * @param [params.condition.httpErrorCodeReturnedEquals] - http code
+    *   that triggers a redirect
```

Comment:
There seems to be a lot more params then are used, is this right?

Reply:
Sure, but it would be better to handle it in a specific ticket. The scope is the migration, we try to keep the code as similar as possible for now.

Reply:
This specific bit of JSDoc looks like an attempt at type hints more than anything, so I'm not surprised it contains more than what we actually use.

What we could do is move the JSDoc to the types (`Redirect` and `Condition`), because it no longer has a lot of meaning directly on the constructor